server/src/test/java/org/elasticsearch/index/mapper/NonDynamicFieldMapperTestCase.java

```diff
@@ -11,114 +11,110 @@
 import org.elasticsearch.action.admin.indices.create.CreateIndexRequestBuilder;
 import org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequestBuilder;
 import org.elasticsearch.common.xcontent.support.XContentMapValues;
-import org.elasticsearch.plugins.MapperPlugin;
-import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.test.ESSingleNodeTestCase;
 import org.elasticsearch.xcontent.XContentType;
 
-import java.io.IOException;
-import java.util.Collection;
 import java.util.List;
-import java.util.Map;
+import java.util.Locale;
 
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
 
-public class DynamicFieldMapperTests extends ESSingleNodeTestCase {
-    @Override
-    protected Collection<Class<? extends Plugin>> getPlugins() {
-        return List.of(NonDynamicFieldPlugin.class);
-    }
+public abstract class NonDynamicFieldMapperTestCase extends ESSingleNodeTestCase {
+
+    protected abstract String getTypeName();
+
+    protected abstract String getMapping();
```

Comment:
Nit: The format of the string returned by this method is quite specific and could trip people up in the future. Would a `Map<String, Object>` return type here work? We could then convert it to JSON in the abstract class. WDYT?

Reply:
I've been thinking about this too - I'd like to have more use cases before reaching a decision about the format. I don't see a particular advantage on using something similar to `MapperTestCase.minimalMapping()`, at least for now.